### ISSUE TITLE
Release macos and linux arm64 wheel files to PyPI

### ISSUE THREAD

**Issue by tboddyspargo** [pymssql/pymssql#763]
**Describe the bug**
Currently, users must built this package from source on macos and linux arm64 platforms requiring special environment variables to be set. This special handling makes it difficult to write concise, reproducible, and portable build scripts and Dockerfiles.

It would significantly reduce container image build times, and build complexity for my team and other members of the community of there was a released wheel of `pymssql` for the `arm64` architecture on both `macos` and `manylinux` (linux) operating systems.

**To Reproduce**
Examples can be seen in https://github.com/pymssql/pymssql/issues/731 and https://github.com/pymssql/pymssql/issues/727

**Expected behavior**
`pip install pymssql` should download and install a compatible wheel file on maxos and linux (e.g. inside a docker container) with Apple Silicon (M1/M2) chip.

**Current Behavior**
`pip install pymssql` reverts to building the wheel file from source since none is available on PyPI.

<details>
<summary>Example Error Logs</summary>

```
Building wheel for pymssql (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pymssql (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [25 lines of output]
      setup.py: platform.system() => Darwin
      setup.py: platform.architecture() => ('64bit', '')
      setup.py: platform.libc_ver() => ('', '')
      setup.py: include_dirs => []
      setup.py: library_dirs => []
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-12.2-arm64-cpython-38
      creating build/lib.macosx-12.2-arm64-cpython-38/pymssql
      copying src/pymssql/__init__.py -> build/lib.macosx-12.2-arm64-cpython-38/pymssql
      running build_ext
      cythoning src/pymssql/_mssql.pyx to src/pymssql/_mssql.c
      cythoning src/pymssql/_pymssql.pyx to src/pymssql/_pymssql.c
      building 'pymssql._mssql' extension
      creating build/temp.macosx-12.2-arm64-cpython-38
      creating build/temp.macosx-12.2-arm64-cpython-38/src
      creating build/temp.macosx-12.2-arm64-cpython-38/src/pymssql
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -I/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX12.3.sdk/usr/include -I/Users/tyler/dev/datachat/venv3.8/include -I/Users/tyler/.pyenv/versions/3.8.13/include/python3.8 -c src/pymssql/_mssql.c -o build/temp.macosx-12.2-arm64-cpython-38/src/pymssql/_mssql.o -DMSDBLIB
      src/pymssql/_mssql.c:699:10: fatal error: 'sqlfront.h' file not found
      #include "sqlfront.h"
               ^~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pymssql
```

</details>

**Context (Environment)**
 - MacOS Monterey, Apple Silicon M1 Max
 - please include the output of the following command:
    `/<path to your python>/python -c "import pymssql; print(pymssql.version_info())"`
    **NOTE: ouptut not available because the package was not installed due to lack of wheel availability and necessary clang libraries.**

**Comment by cwegener** [comment]
Since wheels are built using Github hosted runners, this issue depends on https://github.com/actions/runner-images/issues/2187 before arm64 wheels for macos can be built.

For linux aarch64 wheels though, I would suggest that someone open up a separate issue, since building aarch64 wheels for linux should not need any new features on Github's runner images in order to enable that.

@termim I'd be keen to hear your thoughts. Context: more and more developers (not just in my team) will use Apple's ARM silicon as they replace their old Intel based Macbooks and having a smooth developer experience in projects that use pymssql is important.

**Comment by termim** [comment]
I do not have access to Apple ARM devices to build and test pymssql wheels, so my thoughts are pretty obvious - PRs are very welcome!

**Comment by cwegener** [comment]
> I do not have access to Apple ARM devices
Yes. I think the ability to build macos arm64 wheels fully depends on GH hosted Apple ARM runners becoming available at some point in the future.

I will have a look into Linux aarch64 builds myself and send a PR.

**Comment by evbo** [comment]
@cwegener just curious if you found a way to build on Linux Arm64? I am in the same boat ;)

**Comment by cwegener** [comment]
Not yet! I was just thinking of this topic again yesterday. I'll try and spend some time next weekend working on adding the arm64 linux builds.

**Comment by samh194** [comment]
@cwegener did you get a chance to look at this?

**Comment by cwegener** [comment]
> @cwegener did you get a chance to look at this?

I'm still working on this. I  first want to have a local method of building artefacts and running tests without relying on Github Actions.

I ran into a delay trying to get a local emulation set up. I attempted `proot` to emulate arm64 without needing root permissions, but couldn't get it to work.

I'm now I'm back to using `mkosi` and `systemd-nspawn` for doing testing `arm64` locally which "just works" for doing the emulation and I'm unblocked again.

**Comment by acn-thanapat-sontayasara** [comment]
Hello @cwegener , I was wondering if you might have any updates on this matter.
Your assistance would be greatly appreciated. Thank you. 👍🏻

**Comment by samh194** [comment]
@cwegener Have you managed to get this working?

**Comment by ThorstenEngel** [comment]
@cwegener: in case it is of any help: I personally use Docker (on Windows - regular Intel CPU) and I easily can build (and test) a local Dockerfile for arm64 and amd64 with e.g. this command. But there are other possibilites as well.

- docker buildx build --platform=linux/amd64,linux/arm64 .

**Comment by nafeabd** [comment]
Is there any update on this? It will really help us to have arm64 wheels available.

**Comment by RamaMalladiAWS** [comment]
Installing `freetds-devel` for example resolves this issue.
Installing stable version from source `freetds-dev.1.4.51` https://www.freetds.org/files/current works too.
